framework/include/physics/PhysicsBase.h

```diff
@@ -249,3 +463,177 @@ PhysicsBase::checkVectorParamsNoOverlap(const std::vector<std::string> & param_v
+
+    for (const auto & param_name : parameter_names)
+    {
+      const std::vector<T> & param_vector = getParam<std::vector<T>>(param_name);
+      if (block_names.size() != param_vector.size())
+        paramError(param_name,
+                   "The number of entries in '" + param_name + "' (" +
+                       std::to_string(param_vector.size()) +
+                       ") is not the same as the number of blocks"
+                       " (" +
+                       std::to_string(block_names.size()) + ") in '" + block_param_name + "'!");
+    }
+  }
+  else
+  {
+    unsigned int previous_size = 0;
+    for (unsigned int param_i = 0; param_i < parameter_names.size(); ++param_i)
+    {
+      const std::vector<T> & param_vector = getParam<std::vector<T>>(parameter_names[param_i]);
+      if (param_i == 0)
+      {
+        if (param_vector.size() > 1)
+          paramError(parameter_names[param_i],
+                     "The user should only use one or zero entries in " + parameter_names[param_i] +
+                         " if " + block_param_name + " not defined!");
+        previous_size = param_vector.size();
+      }
+      else
+      {
+        if (previous_size != param_vector.size())
+          paramError(parameter_names[param_i],
+                     "The number of entries in '" + parameter_names[param_i] +
+                         "' is not the same as the number of entries in '" +
+                         parameter_names[param_i - 1] + "'!");
+      }
+    }
+  }
+}
+
+template <typename T>
+bool
+PhysicsBase::parameterConsistent(const InputParameters & other_param,
+                                 const std::string & param_name,
+                                 bool warn) const
+{
+  assertParamDefined<T>(param_name);
+  mooseAssert(other_param.have_parameter<T>(param_name),
+              "This should have been a parameter from the parameters being compared");
+  bool consistent = true;
+  if (parameters().isParamValid(param_name) && other_param.isParamValid(param_name))
+  {
+    if constexpr (std::is_same_v<MooseEnum, T>)
+    {
+      if (!getParam<T>(param_name).compareCurrent(other_param.get<T>(param_name)))
+        consistent = false;
+    }
+    else if (getParam<T>(param_name) != other_param.get<T>(param_name))
+      consistent = false;
+  }
+  if (warn && !consistent)
+    mooseWarning("Parameter " + param_name + " is inconsistent between Physics \"" + name() +
+                 "\" of type \"" + type() + "\" and the parameter set for \"" +
+                 other_param.get<std::string>("_action_name") + "\" of type \"" +
+                 other_param.get<std::string>("action_type") + "\"");
+  return consistent;
+}
+
+template <typename T>
+void
+PhysicsBase::warnInconsistent(const InputParameters & other_param,
+                              const std::string & param_name) const
+{
+  parameterConsistent<T>(other_param, param_name, true);
```

Comment:
I think you should lift the warning from parameter consistent to this guy and let parameter consistent just decide if it is consistent or not.

Reply:
removed the consistency checks. I added that to merge physics over multiple components. I dont know if this will be used in the end